.github/workflows/lint_code.yml

```
@@ -0,0 +1,36 @@
+name: Lint
+
+on:
+  push:
+
+env:
+  TERRAFORM_VERSION: "1.5.7"
+
+jobs:
+  linting:
+    name: Lint Terraform files
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.11.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install Terraform
+        uses: hashicorp/setup-terraform@v2.0.3
+        with:
+          terraform_version: ${{ env.TERRAFORM_VERSION }}
+
+      - name: Run Terraform format
+        run: terraform fmt -recursive -check
```

let's add another step to validate the syntax using Terraform [validate](https://developer.hashicorp.com/terraform/cli/commands/validate) command. That require you to init the terraform project first

```suggestion
      - name: Run Terraform format
        run: terraform fmt -recursive -check
        
      - name: Initialize Terraform
        run: terraform init
      
      - name: Validate Terraform configurations
        run: terraform validate -no-color    
```